pkg/hub/clusterrole/controller.go

```diff
@@ -0,0 +1,92 @@
+package clusterrole
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	clusterv1informer "github.com/open-cluster-management/api/client/cluster/informers/externalversions/cluster/v1"
+	clusterv1listers "github.com/open-cluster-management/api/client/cluster/listers/cluster/v1"
+	"github.com/open-cluster-management/registration/pkg/helpers"
+	"github.com/open-cluster-management/registration/pkg/hub/clusterrole/bindata"
+
+	"github.com/openshift/library-go/pkg/assets"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	operatorhelpers "github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	"k8s.io/apimachinery/pkg/labels"
+	rbacv1informers "k8s.io/client-go/informers/rbac/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const manifestDir = "pkg/hub/clusterrole"
+
+var clusterRoleFiles = []string{
+	"manifests/managedcluster-registration-clusterrole.yaml",
+	"manifests/managedcluster-work-clusterrole.yaml",
+}
+
+// clusterroleController maintains the necessary clusterroles for registraion and work agent on hub cluster.
+type clusterroleController struct {
+	kubeClient    kubernetes.Interface
+	clusterLister clusterv1listers.ManagedClusterLister
+	eventRecorder events.Recorder
+}
+
+// NewManagedClusterClusterroleController creates a clusterrole controller on hub cluster.
+func NewManagedClusterClusterroleController(
+	kubeClient kubernetes.Interface,
+	clusterInformer clusterv1informer.ManagedClusterInformer,
+	clusterRoleInformer rbacv1informers.ClusterRoleInformer,
+	recorder events.Recorder) factory.Controller {
+	c := &clusterroleController{
+		kubeClient:    kubeClient,
+		clusterLister: clusterInformer.Lister(),
+		eventRecorder: recorder.WithComponentSuffix("managed-cluster-clusterrole-controller"),
+	}
+	return factory.New().
+		WithInformers(clusterInformer.Informer(), clusterRoleInformer.Informer()).
```

Comment:
adding filter here will be better.
I think you also need clusterinformer?

When it is triggered when we create a new cluster?
When it is triggered when there are no clusters.

Reply:
OK, I will adding a filer here and I already added the `clusterinformer`